### DISCUSSION TITLE
How to configure Cron job with Synology Container Manager and Reverse Proxy?

### DISCUSSION THREAD

**Discussion by Sapperloettchen** [firefly-iii/firefly-iii#8876] [Firefly III]
⁸Hi there dear community

So happy that i finally found firefly!! Just amazing what you created here James, thank you very much! 
Everythink works fine behind a reverse proxy where i mapped custom ports to firefly and the importer to make it reachable outside my home network. But just one thing making me headache since two weeks. I can't get cron running... i tried everythink, including asking dear Chatpgt ;). 

This is my setup: 
- Synology with Container Manager (formerly Docker)
- I have my own domainname and use a reverse proxy for firefly. Firefly and the importer are working fine. 
- I use the standard cron phrase:
  cron:
    image: alpine
    restart: always
    container_name: firefly_iii_cron
    command: sh -c "echo \"* * * * * wget -qO- https://firefly.[mydomainname].com/api/v1/cron/jpPT4E6m2nCWz4P4wWaAVdeMRDBikZQi\" | crontab - && crond -f -L /dev/stdout"
    networks:
      - firefly_iii

The STATIC_CRON_TOKEN is a 32 long string in the .env file. like described in the documentation. I also tried to put the 32 characters directly in the line but it changed nothing.

I tried to completely disable the routers and synology firewall in case it's a blocked portnumber or so... changed nothing. I gave permissions to all users on the folders, changed nothing.

The error i get when leave the firewall on:
wget: can't connect to remote host (my public ip): Operation timed oute
,stdout,crond: user root: process already running: wget -qO- https://firefly.[mydomainname].com/api/v1/cron/jpPT4E6m2nCWz4P4wWaAVdeMRDBikZQi

The stange thing is, that when i paste the URL directly into my browswer, it works!! The transaction was made. But why is it working in the browser but not when executed via the normal docker-compose.yml?
This was the URL that worked: https://firefly/.[mydomainname].com/api/v1/cron/jpPT4E6m2nCWz4P4wWaAVdeMRDBikZQi

If anyone has an idea, it would mean me a lot, really 😊.

**Comment by JC5** [comment]
Alright, so it tries to run but it can't reach the IP address. I don't know a lot about Synology, but I expect that you need to point to the internal IP plus port, so something like `172.16.0.1:8080`. Which IP address I can't be sure, but I know you need to add `:8080`.

**Reply by Sapperloettchen** [reply]
Thanks a lot for the reply! I tried a few different IP's but none of them worked, unfortunately. But i could solve it with another approach. I made a job with the native built-in synology task scheduler. It calls once a day my puplic firefly adress like described above and this works fine.

**Reply by JC5** [reply]
That's a great solution! I'm glad you found one. Thanks for the support, I appreciate it! 😁

**Reply by Sapperloettchen** [reply]
Hi James, hope you're fine and well. 
Sorry for another question. The cron job worked fine liked described above until the new firefly update recenty. Now, i don't know exactly why, the cron job doesn't work even when i paste the URL directly in the browser.... all i get is this. Do you have an idea what is different now? Thanks

![Screenshot 2024-05-24 082416](https://github.com/firefly-iii/firefly-iii/assets/55541491/634a0b5f-b024-4509-80e9-3d05f86734d6)

![Screenshot 2024-05-24 083742](https://github.com/firefly-iii/firefly-iii/assets/55541491/33d0f657-3c15-4d3a-8788-1b5175cf5210)

**Reply by Sapperloettchen** [reply]
I just played around a litte bit and now downgraded to the previous version 6.1.15 and the cron tasks work perfectly. So i guess, something changed in 6.1.16 that the cron task is not working anymore? I can go with the previous version but it would be great if future updates work with cron again :).

![Screenshot 2024-05-24 123735](https://github.com/firefly-iii/firefly-iii/assets/55541491/fadb30e6-59cc-4750-a371-ca16566ee1c6)

**Reply by JC5** [reply]
Nothing changed in between those versions though. And if the count is indeed "0 seconds since the previous run", it simply won't run again.

**Reply by Sapperloettchen** [reply]
Thanks for the reply. You're right, know the problem occured also in the previous version, the same thing like described above in version 16. 
I did another research and found out, that HSTS is the problem... but I'm not sure why. Do you know something about that? The .wget-hsts file is automatically created by the cron job i guess and i see it, but i don't have a clue what to do in this case... I gave all user read and write rights to the file but that changed nothing.
 
![Screenshot 2024-05-27 082754](https://github.com/firefly-iii/firefly-iii/assets/55541491/1a10a91f-2dbb-4fb5-9aaa-bd05afc9ac29)

**Reply by JC5** [reply]
Mmm, that file isn't mine I'm afraid. Is that something locally?
